### PR TITLE
[SPARK-47868][CONNECT] Fix recursion limit error in SparkConnectPlanner and SparkSession

### PR DESCRIPTION
**Pull request by tomvanbussel**
### What changes were proposed in this pull request?

This PR adds a helper function to `ProtoUtils` that calls a proto parser on a byte array with an increased recursion limit. This helper function is used to enhance the `parseFrom` calls in `SparkSession` and `SparkConnectPlanner`.

### Why are the changes needed?

Otherwise Spark Connect extensions will get the following exception:
```
org.apache.spark.SparkException: grpc_shaded.com.google.protobuf.InvalidProtocolBufferException: Protocol message had too many levels of nesting.  May be malicious.  Use CodedInputStream.setRecursionLimit() to increase the depth limit.
```

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests to make sure nothing breaks. Manually tested (using an extension that is currently in development) that this solves the exceptions.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by grundprinzip — APPROVED** [review]

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -586,9 +587,13 @@ class SparkSession private[sql] (
 
   @DeveloperApi
   def execute(extension: Array[Byte]): Unit = {
+    val any = ProtoUtils.parseWithRecursionLimit(
+      extension,
+      com.google.protobuf.Any.parser(),
+      recursionLimit = 4096)
```
Comment (tomvanbussel) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala:593]: Is there a local config for this available somewhere?

Comment (vicennial) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala:593]: How about in https://github.com/apache/spark/blob/master/connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/config/ConnectCommon.scala#L19-L22?

Comment (tomvanbussel) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala]: Thanks! That at least removes the duplication. Ideally we would be able to make this configurable though.

Comment (vicennial) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala]: The concept of local confs doesn't really exist on the client side per se, configurability is instead achieved through the client builder like so - https://github.com/apache/spark/blob/master/connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala#L692-L706

**Review by vicennial — APPROVED** [review]
LGTM!

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**